packages/gestalt/src/AvatarFoundation.js

```diff
@@ -0,0 +1,116 @@
+// @flow strict
+import { type Node } from 'react';
+import classnames from 'classnames';
+import Box from './Box.js';
+import typography from './Typography.css';
+import { useColorScheme } from './contexts/ColorScheme.js';
+import icons from './icons/index.js';
+import styles from './Icon.css';
+import colors from './Colors.css';
+import avatarStyles from './AvatarGroup.css';
+
+type ResponsiveFitSizeBoxProps = {| children: Node, outline: boolean |};
+
+function ResponsiveFitSizeBox({ children, outline }: ResponsiveFitSizeBoxProps): Node {
+  const { colorGray0 } = useColorScheme();
+
+  return (
+    <Box
+      color="lightGray"
+      dangerouslySetInlineStyle={{
+        __style: {
+          paddingBottom: '100%',
+          boxShadow: outline ? `0 0 0 1px ${colorGray0}` : undefined,
+        },
+      }} // When specifying a padding by percentage, it's always based on the width of the parent container so we get a property that's equal to the width.
+      position="relative"
+      rounding="circle"
+    >
+      <Box
+        position="absolute"
+        // top left bottom right constrains the circle to the exact dimensions of the responsive parent square
+        top
+        left
+        bottom
+        right
+        display="flex"
+        justifyContent="center"
+      >
+        {children}
+      </Box>
+    </Box>
+  );
+}
+
+type Props = {|
+  children?: string | number,
+  fontSize?: string,
+  outline?: boolean,
+  textAnchor?: 'start' | 'middle' | 'end',
+  title?: string,
+  translateX?: 'translateX10' | 'translateX15',
```

Comment:
This is a bit repetitive. Can probably either change the name to just `translate` or remove the `translateX` prefixes

Reply:
DONE